modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/DiscoverAdapter.kt

```diff
@@ -502,18 +505,25 @@ internal class DiscoverAdapter(
                     )
                 }
                 is CategoriesRedesignViewHolder -> {
-                    val adapter = CategoriesListRowAdapter(listener::onPodcastListClicked)
+                    var discoverCategories = emptyList<DiscoverCategory>()
+                    val adapter = CategoriesListRowRedesignAdapter(
+                        onPodcastListClick = listener::onPodcastListClicked,
+                        onAllCategoriesClick = { listener.onAllCategoriesClicked(discoverCategories) },
+                    )
```

Comment:
It'd be nice to use the same patterns I mentioned for previous adapters but I don't think it can be easily done given that is already existing adapter with some logic.

Reply:
yeah. we have this pattern for this whole class. I would prefer to address this in another PR